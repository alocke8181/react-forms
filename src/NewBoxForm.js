import React, {useState} from "react";

const NewBoxForm = ({addBox}) =>{
    const [formData, setFormData] = useState({
        bgColor : '#000000',
        width : 0,
        height : 0
    })

    const handleChange = (e) =>{
        const {name, value} = e.target;
        setFormData((data) =>({
            ...data,
            [name]: value
        }));
    };

    const handleSubmit = (e) =>{
        e.preventDefault();
        addBox(formData.bgColor, formData.width, formData.height);
        setFormData({
            bgColor : '#000000',
            width : 0,
            height : 0
        });
    }

    return (
        <form onSubmit={handleSubmit}>
            <label htmlFor="bgColor">Background Color</label>
            <input
                id="bgColor"
                type="color"
                name="bgColor"
                placeholder="#000000"
                value={formData.bgColor} 
                onChange={handleChange}
            />
            <label htmlFor="width">Width px</label>
            <input
                id="width"
                type="number"
                name="width"
                placeholder="100"
                value={formData.width}
                onChange={handleChange}
            />
            <label htmlFor="height">Height px</label>
            <input
                id="height"
                type="number"
                name="height"
                placeholder="100"
                value={formData.height}
                onChange={handleChange}
            />
            <button id="box-submit">Submit</button>
        </form>
    )
}

export default NewBoxForm;