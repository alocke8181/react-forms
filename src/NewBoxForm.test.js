import { render } from '@testing-library/react';
import NewBoxForm from './NewBoxForm';

const testAddBox = () =>{console.log('test')}

test('smoke',()=>{
    render(<NewBoxForm addBox={testAddBox} />)
})

test('snapshot',()=>{
    const {asFragment} = render(<NewBoxForm addBox={testAddBox} />)
    expect(asFragment()).toMatchSnapshot();
})