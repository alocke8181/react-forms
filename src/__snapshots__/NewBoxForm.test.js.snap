// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="bgColor"
    >
      Background Color
    </label>
    <input
      id="bgColor"
      name="bgColor"
      placeholder="#000000"
      type="color"
      value="#000000"
    />
    <label
      for="width"
    >
      Width px
    </label>
    <input
      id="width"
      name="width"
      placeholder="100"
      type="number"
      value="0"
    />
    <label
      for="height"
    >
      Height px
    </label>
    <input
      id="height"
      name="height"
      placeholder="100"
      type="number"
      value="0"
    />
    <button
      id="box-submit"
    >
      Submit
    </button>
  </form>
</DocumentFragment>
`;
